packages/starknet-snap/src/declareContract.ts

```diff
@@ -1,30 +1,48 @@
+import { heading, panel, DialogType } from '@metamask/snaps-sdk';
+
+import type { ApiParams, DeclareContractRequestParams } from './types/snapApi';
+import { logger } from './utils/logger';
 import { toJson } from './utils/serializer';
-import { ApiParams, DeclareContractRequestParams } from './types/snapApi';
-import { getNetworkFromChainId, getDeclareSnapTxt, showAccountRequireUpgradeOrDeployModal } from './utils/snapUtils';
+import {
+  getNetworkFromChainId,
+  getDeclareSnapTxt,
+  showAccountRequireUpgradeOrDeployModal,
+} from './utils/snapUtils';
 import {
   getKeysFromAddress,
   declareContract as declareContractUtil,
   validateAccountRequireUpgradeOrDeploy,
 } from './utils/starknetUtils';
-import { heading, panel, DialogType } from '@metamask/snaps-sdk';
-import { logger } from './utils/logger';
 
+/**
+ *
+ * @param params
+ */
 export async function declareContract(params: ApiParams) {
   try {
     const { state, keyDeriver, requestParams, wallet } = params;
     const requestParamsObj = requestParams as DeclareContractRequestParams;
 
     logger.log(`executeTxn params: ${toJson(requestParamsObj, 2)}}`);
 
-    const senderAddress = requestParamsObj.senderAddress;
+    const { senderAddress } = requestParamsObj;
     const network = getNetworkFromChainId(state, requestParamsObj.chainId);
-    const { privateKey, publicKey } = await getKeysFromAddress(keyDeriver, network, state, senderAddress);
+    const { privateKey, publicKey } = await getKeysFromAddress(
+      keyDeriver,
```

Comment:
Why we don't do : `keyDeriver as unknown as BIP44AddressKeyDeriver,` here ?

Reply:
it was because getKeysFromAddress is using any for keyDeriver, i will change back